index.html

```diff
@@ -1615,6 +1618,179 @@ <h3>
+              [=iteration/continue=].
+            </li>
+            <li>If <var>entry</var>["value"] is not a valid [=URL=], 
+              [=iteration/continue=].
+            </li>
+            <li>Set <var>entry</var>["value"] to its own [=origin=].
+            </li>
+            <li>Validate |entry| using the steps from [=validate scope 
+              extensions=], given [=ordered map=] |entry:ordered map|, and 
+              [=URL=] |resolved_scope:URL|.
+            </li>
+            <li>If validation failed, [=iteration/continue=].
+            </li>
+            <li>
+              Set |entry|["scope"] to |resolved_scope|.
+            </li>
+            <li>[=list/Append=] <var>entry</var> to 
+              <var>processedScopeExtensions</var>.
+            </li>
+          </ol>
+        </li>
+      </ol>
+      <aside class="note">
+        `origin` is the only type of scope extension entry that is 
+        currently specified. Other types may be added in the future to
+        support different use cases. 
+      </aside>
+      </section>
+      <section>
+      <h2>Validating scope extensions</h2>
+      <p>
+        To <dfn>validate scope extension</dfn>, given [=ordered map=] 
+        |extension:ordered map|, [=URL=] |manifest_id:URL| and [=URL=] 
+        |resolved_scope:URL|:
+      </p>
+      <ol class="algorithm">
+        <li>Set |resolved_scope| to empty URL.</li>
+        <li>Let |download_url| be the result of constructing 
+          [=web-app-origin-association=] URL from the [=origin=] in 
+          |extension|["value"].
+        </li>
+        <li>Let [=ordered map=] |json:ordered map| be the result of downloading
+          |download_url|. 
+        </li>
+        <li>If the download fails, return false.
+        </li>
+        <li>If |json|[|manifest_id|] does not exist or is not an 
+          [=ordered map=], return false.</li>
+        <li>Let [=string=] |scope_path:string| be "/".</li>
+
+        <li>If |json|[|manifest_id|]["scope"] exists and is [=string=], 
+          set |scope_path| to |json|[|manifest_id|]["scope"]. 
+        </li>
+        <li>Set |resolved_scope:URL| be the result of parsing |scope_path| with 
+          |extension|["value"] as the base URL.
+        </li>
+        <li>return true.</li>
+      </ol>
+      <h3>The web-app-origin-association file</h3>
+      <p>A <dfn>web-app-origin-association</dfn> file is a JSON file that can be
+        used to associate the origin it is in with one or more web applications.
+        It does this by referencing manifest [=manifest/ids=]. 
+      </p>
+      <p>
+        A [=web-app-origin-association=] file is expected to be downloadable
+        from <code>[origin]/.well-known/web-app-origin-association</code>.
+      </p>
+    </section>
+    <section>
+      <h2>Applying scope extensions</h2>
+      <p>The user agent MAY apply extended scope to some features but 
+        not others.
+      </p>
```

Comment:
We should `dfn` the application, and define the algorithm? Ideally we can reference the `[=manifest/within scope=]` algorithm but give it different 'input arguments', as we have the same 'scope' member here (and that's what we want in the future - same algorithm, but different inputs.

Reply:
I added a new dfn at the top of the whole section to specify that extension scope is original manifest scope + URLs in extensions. Does that accomplish the same thing? 

I'm using "Application" section for things like UA should keep users informed of the security origin, etc.